bridges/RedditBridge.php

```diff
@@ -0,0 +1,59 @@
+<?php
+class RedditBridge extends FeedExpander {
+
+	const MAINTAINER = 'leomaradan';
+	const NAME = 'Reddit Bridge';
+	const URI = 'https://www.reddit.com/';
+	const DESCRIPTION = 'Reddit bridge for Feedly';
+
+	const PARAMETERS = array(
+		'single' => array(
+			'r' => array(
+				'name' => 'SubReddit',
+				'required' => true,
+				'exampleValue' => 'selfhosted',
+				'title' => 'SubReddit name'
+			)
+		),
+		'multi' => array(
+			'rs' => array(
+				'name' => 'SubReddits',
+				'required' => true,
+				'exampleValue' => 'selfhosted, php',
+				'title' => 'SubReddit names, separated by commas'
+			)
+		)
+	);
+
+	public function collectData(){
+
+		$r = $this->getInput('r');
+		$rs = $this->getInput('rs');
+
+		$subreddits = [];
+
+		if($r !== null) {
+			$subreddits[] = $r;
+		}
+
+		if($rs !== null) {
+			$split = explode(',', $rs);
+			$subreddits = array_merge($split, $subreddits);
+		}
```

Comment:
Instead of checking each parameter individually you can also do this:

```php
switch($this->queriedcontext) {
    case 'single': $subreddits[] = $this->getInput('r'); break;
    case 'multi': $subreddits = explode(',', $this->getInput('rs')); break;
}
```

RSS-Bridge guarantees that only one context is fulfilled at any given time.